feature: add toolbar buttons via plugins

rolandlo opened the pull request:
Fixes #4669. Fixes #2936. Fixes #2431. Also addresses #3669.

This PR allows registering toolbar buttons (with any ID and any icon in the icon theme) that can be added to any toolbar.

These custom tools (executing plugin callback functions) can be added to the toolbar either by customizing the toolbar in the GUI or by editing `toolbar.ini`. 

The plugin adds the toolbar via e.g. 
```lua
app.registerUi({callback ="blueDashedPen", toolbarId="CUSTOM_PEN_1", iconName="bluePenIcon"})
```

It's also possible to combine adding a toolbar button with adding a menu entry:

```lua
app.registerUi({["menu"] = "Mirror strokes horizontally", ["callback"] = "cb", ["accelerator"] = "<Control>t", ["toolbarId"] = "MIRROR", ["iconName"] = "view-mirror-symbolic"})
```

Here `view-mirror-symbolic` is an icon that actually exists. I found via Gnome's symbol library (it's part of the Adwaita icon theme). 

![grafik](https://user-images.githubusercontent.com/40485433/220704573-b341e663-ba75-4959-8fe6-7f8138072aa1.png)


A little demo (see the icon on the top left, right below the `File` menu):

https://user-images.githubusercontent.com/40485433/220704181-bbcaa122-db17-4366-b91a-fc6aec1cc06d.mp4

bhennion commented:
That's very nice!
I have only glanced at the code, but I have a question/concern: giving the user the ability to create new toolbarId's (the strings used in toolbar.ini), we could be opening the door to some form of spoofing: create a nasty plugin with toolbarId being, say, `SAVE` (and matching icon).

What happens then? Are we guarantied that when we click on the SAVE button, it saves and does not execute the plugin's callback?
I see two ways of ensuring that:
1. Making certain that stock buttons are prefered to plugin buttons in case of competing toolbarId's + issueing an error if this happens
2. Adding a PLUGIN:: namespace to all toolbarIds created via the plugin interface. (so basically add a `toolbarId = "PLUGIN::" + toolbarId;` somewhere).

I'd say 2. would be better, safer and likely to get unwillingly destroyed by our future selves.
The option 1. also has the drawbacks
- Future bug reports "I can't get my plugin to create a button" due to the author not realizing the toolbarId was already taken.
- if an update adds new stock toolbar buttons, and the added toolbarIds then break some user's plugin.

rolandlo commented:
I like your second solution and implemented it. Even from a user perspective it's cleaner looking at `toolbar.ini`.
Talking about security of the plugin system, we should also ensure that all plugins loaded from the `.config` directory are disabled by default.

rolandlo commented:
Rebased on current master and squashed the commits related to the actual feature.

Dropping the second commit (test plugin) and merging the first commit in 48 hours if no objections are raised.
